### ISSUE TITLE
Autocomplete ADA: mat-option announcement lags behind by one entry

### ISSUE THREAD

**Issue by erickcchan** [angular/components#19745]
**Summary:**
Autocomplete options screen reader announcement lags behind by one entry.

**Steps to reproduce:**
1. Turn on VoiceOver on Safari on Mac.
2. Access a page with the autocomplete field, e.g.: https://material.angular.io/components/autocomplete/examples
3. Use the keyboard to access the field, and press up/down arrows to cycle through the items.

**Expected behavior:**
* Items are announced as they are visited

**Actual behavior:**
* Seems like it always announces the previously visited (highlighted) item.

**Impact:**
This is a showstoppper bug for ADA compliance, as a blind user will not be aware that they've picked the wrong item.

**Browsers:**
* Observed that Chrome on VoiceOver also has the issue
* Observed that Firefox on VoiceOver does NOT have the issue.

**Comment by crisbeto** [comment]
It seems to be working for me both on Firefox and Chrome. Are you seeing it on a specific example or on all of them?

**Comment by erickcchan** [comment]
Hi crisbeto.  I'm seeing the issue with Safari and Chrome with VoiceOver.  I understand maybe VoiceOver is not fully supported by Chrome, but it should support Safari right? Did you try Safari on Mac?

**Comment by crisbeto** [comment]
I just tried it out against Safari 13.1 and it seemed to be working correctly. There was one case where moving to a different option while the previous one was being announced didn't announce the new option, but I don't think we can do much about it.

**Comment by loraruffin** [comment]
I am seeing the same error in Chrome Version 85.0.4183.121. However it is working as expected in Safari Version 13.1.2.
Dropdown items are off by 1 when announced.

**Comment by jelbourn** [comment]
We haven't been able to reproduce this issue. Feel free to open a new issue if you are able to provide a reproduction via StackBlitz.

(this is a canned response)
